app/models/customer.rb

```diff
@@ -35,6 +35,7 @@ class Customer < ApplicationRecord
                     uniqueness: { scope: :enterprise_id, message: I18n.t('validation_msg_is_associated_with_an_exising_customer') }
 
   scope :of, ->(enterprise) { where(enterprise_id: enterprise) }
+  scope :managed_by, ->(user) { user&.persisted? ? where(user: user).or(of(user.enterprises)) : none }
```

Comment:
> It seems unlikely that user is nil or not persisted, and it would probably return an empty result anyway I think.

I also thought the same, but after running the tests I discovered that when a user calls the API v1 without being authenticated, a `NullObject` will be used as the current_user. 
`Customer.manager_by(currect_user)` will return all the customer with `user_id = nil`

Reply:
>if the user is a customer of their own enterprise, they should be able to access their customer. But if not, I don't think they should need to.

When can we say that a user is a customer? Is it when customer.user = user?
 
I did some research and from what I understand a user can create multiple customers. Does this mean that every customer represents the user?